pkg/kv/kvserver/batcheval/cmd_delete_range_test.go

```diff
@@ -0,0 +1,200 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package batcheval_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval"
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/settings/cluster"
+	"github.com/cockroachdb/cockroach/pkg/storage"
+	"github.com/cockroachdb/cockroach/pkg/storage/enginepb"
+	"github.com/cockroachdb/cockroach/pkg/util/hlc"
+	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/errors"
+	"github.com/stretchr/testify/require"
+)
+
+// TestDeleteRangeTombstone tests DeleteRange range tombstones directly, using
+// only a Pebble engine.
+//
+// Most MVCC range tombstone logic is tested exhaustively in the MVCC history
+// tests, this just tests the RPC plumbing.
+func TestDeleteRangeTombstone(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	// Initial data for each test: point keys at b@2, c@5, d@2, point tombstone at
+	// d@3, point intent at i@7, and a range tombstone at [k-p)@3.
+	writeInitialData := func(t *testing.T, ctx context.Context, rw storage.ReadWriter) {
```

Comment:
nit, ignore if you think it doesn't help, but I find it easier to reason about test cases when I can easily map deleted range onto the key-time space
```
    //  ts
    // 7|                 i
    // 6|
    // 5|     +
    // 4|
    // 3|       -             o-----o
    // 2|   +   +
    // 1|
    // -+---------------------------------- key
    //  | a b c d e f ... i j k ... p ... z
```

Reply:
Yep, done.